flink-runtime/src/main/java/org/apache/flink/runtime/state/CheckpointableKeyedStateBackend.java

```diff
@@ -45,5 +46,6 @@
      * write out a savepoint in the common/unified format.
      */
     @Nonnull
-    SavepointResources<K> savepoint() throws Exception;
+    SavepointResources<K> savepoint(long checkpointId, CheckpointOptions checkpointOptions)
```

Comment:
Is it really necessary to have the two arguments in this method? Especially the `CheckpointOptions`? Can't we build the SavepointResources in a way that we don't need these parameters?

Reply:
Creating snapshot for checkpoint and for savepoint differently would complicate `HeapSnapshotResources`.
OTH, both these arguments seem relevant to both checkpoint and savepoints. 
What is your concern about these arguments?